### ISSUE TITLE
Document how a contributor can easily contribute to the documentation

### ISSUE THREAD

**Issue by MarcSkovMadsen** [holoviz/panel#4213]
I would like to contribute improvements to the home page (`index.md`) of Panel.

In order to be able to work with the page I need to be able to get visual feedback on the look of the page. Unfortunately my editor VS Code cannot show the `.md` file as it very often can due to special directives.

I'm struggling as there is no documentation on how to get things installed and establish an efficient developer workflow up for the docs.

I've now spend 30 minutes trying to get sphinx and what ever it takes installed. And also be able to build the documentation and establish a fast feedback cycle.

If I run `sphinx-build` or `sphinx-autobuild` it builds for ever. And for some reason also with lots of errors.

I really want to do something about the home page. I believe the current state is simply no longer representative of what Panel is and users would get a very wrong impression of it.

**Comment by philippjfr** [comment]
We have just been discussing better developer docs internally and you should see action on providing much simpler project specific pip and conda based instructions for setting up development environments very soon.

**Comment by cdeil** [comment]
I had the same question - how to set up an environment where I can build and check the docs?

I looked at https://panel.holoviz.org/developer_guide/index.html but there docs build currently isn't mentioned.

For now, trying to follow the instructions I ran into an unrelated error with the current dev env setup: #4248

While reading the docs I noticed a few cases of broken formatting or links and want to send a PR, but before doing so would like to check locally if the edits actually fix the formatting/links or not.

Thank you!

**Comment by cdeil** [comment]
Looking at Panel again, possibly interested to fix some docs formatting while reading around.

However, https://pyviz-dev.github.io/panel/developer_guide/index.html still doesn't mention how to build the docs.

There seems to be an unused file extensions.md ?

Related: #4544

**Comment by philippjfr** [comment]
Indeed, I'm working on this right now in fact, hope to have something by Monday.
